website/client/js/services/userServices.js

```diff
@@ -103,8 +104,28 @@ angular.module('habitrpg')
 
           return Tasks.getUserTasks();
         })
+        // refresh all but completed todos
         .then(function (response) {
-          var tasks = response.data.data;
+          tasks = response.data.data;
+
+          // only refresh completed todos if the user has the completed tabs list open
+          var taskButton = document.getElementById('completedTasksButton');
```

Comment:
I was able to do this but using the list.view == 'complete' in the expression results in it evaluating to true even when on a different page (like Social or Inventory). This results in the call for loading completed tasks being made anytime the user left the task page with the "Done" todos list item selected.